pkg/runctl/controller_test.go

```diff
@@ -47,7 +48,11 @@ func Test_meterAllPipelineRunsPeriodic(t *testing.T) {
 		runctltesting.FakeClusterRole(),
 	)
 
-	c := NewController(cf, ControllerOpts{})
+	c := NewController(
+		klog.FromContext(context.Background()),
```

Comment:
Should use [`ktesting.NewLogger()`](https://pkg.go.dev/k8s.io/klog/v2/ktesting#NewLogger) instead, so that the log output is tied to the test case.